include/clang/AST/Type.h

```diff
@@ -3689,6 +3693,35 @@ class UnresolvedUsingType : public Type {
 };
 
 
+class TypeVariableType : public Type, public llvm::FoldingSetNode {
```

Comment:
Please add a classof method to this class.   That is necessary for Clang's/LLVM hand-implemented dynamic type tests to work properly.